src/SmartComponents/AddSystemModal/AddSystemModal.js

```diff
@@ -123,8 +123,26 @@ export class AddSystemModal extends Component {
         toggleAddSystemModal();
     }
 
+    findNotInComparison(basketContent, comparedContent) {
+        if (comparedContent.length === 0) {
+            return basketContent;
+        } else {
+            return basketContent.filter(item => comparedContent.find(({ id }) => id !== item.id));
+        }
+    }
+
     cancelSelection() {
-        const { toggleAddSystemModal } = this.props;
+        const { baselines, handleBaselineSelection, handleHSPSelection, handleSystemSelection, historicalProfiles,
+            selectBaseline, selectedBaselineContent, selectedHSPContent, selectedSystemContent, selectHistoricProfiles,
+            systems, toggleAddSystemModal } = this.props;
+
+        handleSystemSelection(this.findNotInComparison(selectedSystemContent, systems), false);
```

Comment:
Observation: The system selection (number) on the bulk select is visible for a brief moment when reopening the modal. Then is reset.